Automattic/jetpack
# AI Logo Generator: fix upgrade links

Fixes #38596.

## Proposed changes:

* Fix the internal `useCheckout` hook to work for any site type
* Update the Generator Modal compoment to use the hook instead of try to build the URL on its own
* **Note:** there is a refactoring opportunity here to use the useAICheckout hook already used by some of the other AI tools, but this will demand moving it to the `ai-client` package, updating the imports over the code and then reusing it here, what is a big change; I added a draft task for it, so we can tackle it when we see this tool gaining traction

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






There are 4 scenarios to test here:

* The site has a free plan
* The site has a paid plan and does not have enough requests to generate logos, with two options:
   * it has a logo history
   * it does not have a logo history
* The site has a paid plan and has enough requests to generate logos
   * having a logo history does not matter in this case

We need to simulate the scenarios and the best way to do it is using the backend filters that change the active tier of a site; the snippets for the test scenarios are provided at the end of the testing instructions

For any scenario:

* the upgrade link on Jetpack sites points to the product interstitial
* the upgrade link on Simple sites points to the checkout page

So it is important to test it on a Jetpack site and a Simple site.

Expected behavior on the free plan:

- set your site to simulate the free plan
- go to the block editor, add a site logo block and click the AI extension button
- an alert message will show up
- the upgrade button should point to the right page depending on the site type (see above) and should open on a new tab

<img width="400" alt="Screenshot 2024-07-29 at 15 55 26" src="https://github.com/user-attachments/assets/c3fab5ea-408a-4a1a-bab1-65121d892c14">

---

Expected behavior on the paid plan with enough requests:

- set your site to simulate a paid plan plan with enough requests
- go to the block editor, add a site logo block and click the AI extension button
- the tool should show up correctly (maybe it will generate the first logo if that is the first time you open it)
- only a small upgrade link should be visible under the prompt field, it should point to the right page and should open on a new tab

<img width="500" alt="Screenshot 2024-07-29 at 15 56 58" src="https://github.com/user-attachments/assets/126b6042-5667-4e21-84f6-478962436142">

---

Expected behavior on the paid plan without enough requests and with a logo history

- set your site to simulate a paid plan plan without enough requests
- go to the block editor, add a site logo block and click the AI extension button
- the tool should show up correctly, showing you logo history
- you should see the upgrade nudge under the prompt field, it should point to the right page and should open on a new tab

<img width="500" alt="Screenshot 2024-07-29 at 15 58 15" src="https://github.com/user-attachments/assets/de8f9de2-6158-4949-ba80-ab2c659577a8">

---

Expected behavior on the paid plan without enough requests and without a logo history

- set your site to simulate a paid plan plan without enough requests
- clean the logo generation history by deleting the `logo-history-SITE_ID` key on the local storage
- go to the block editor, add a site logo block and click the AI extension button
- an alert message will show up, similar to the free one
- the upgrade button should point to the right page and should open on a new tab

<img width="500" alt="Screenshot 2024-07-29 at 15 59 11" src="https://github.com/user-attachments/assets/ae2c7502-117a-4901-8027-6dee3afc4370">

---

Test it on a Jetpack site and a Simple site.

### Simulating the tiers

Use the following snippets on your `0-sandbox.php` file to simulate you are using different tiers.

_Free plan:_

```
add_filter( 'jetpack_ai_current_period_requests_count', function() { return 0; } );
add_filter( 'jetpack_ai_all_time_requests_count', function() { return 0; } );
add_filter( 'jetpack_ai_tier_licensed_quantity', function() { return 0; } );
```

_Paid plan with enough requests:_

```
add_filter( 'jetpack_ai_current_period_requests_count', function() { return 0; } );
add_filter( 'jetpack_ai_all_time_requests_count', function() { return 0; } );
add_filter( 'jetpack_ai_tier_licensed_quantity', function() { return 100; } );
```

_Paid plan without enough requests:_

```
add_filter( 'jetpack_ai_current_period_requests_count', function() { return 95; } );
add_filter( 'jetpack_ai_all_time_requests_count', function() { return 95; } );
add_filter( 'jetpack_ai_tier_licensed_quantity', function() { return 100; } );
```

Remember to sandbox the `public-api` URL on your testing machine.
